server/py/services/api/db/sqldb/db.py

```diff
@@ -5656,6 +5656,106 @@ def _enrich_notification(_notification):
             for cooldown, notification in zip(cooldowns, notifications)
         ]
 
+    @staticmethod
+    def create_partitions(
+        session: Session,
+        table_name: str,
+        partitioning_information_list: list[tuple[str, str]],
+    ):
+        """
+        Creates partitions in the specified database table.
+
+        :param session: SQLAlchemy session for database connection.
+        :param table_name: Name of the table where partitions will be created.
+        :param partitioning_information_list: List of tuples, each containing:
+            - partition_name: The name for the partition.
+            - partition_value: The "LESS THAN" boundary value for the partition.
+        """
+        query = text("""
+            SELECT PARTITION_DESCRIPTION
+            FROM INFORMATION_SCHEMA.PARTITIONS
+            WHERE TABLE_NAME = :table_name
+        """)
+
+        existing_partition_values = {
+            row["PARTITION_DESCRIPTION"]
+            for row in session.execute(query, {"table_name": table_name})
+        }
+
+        # Filter partitions to add only those that are not in the table
+        new_partitions = [
+            f"PARTITION p{partition_name} VALUES LESS THAN ({partition_value})"
+            for partition_name, partition_value in partitioning_information_list
+            if str(partition_value) not in existing_partition_values
+        ]
+
+        if not new_partitions:
+            return
+
+        alter_table_template = f"""
+            ALTER TABLE {table_name}
+            ADD PARTITION (
+                {", ".join(new_partitions)}
+            );
+        """
+
+        session.execute(text(alter_table_template))
+        session.commit()
+
+    @staticmethod
+    def drop_partitions(
+        session: Session,
+        table_name: str,
+        cutoff_partition_name: str,
+    ):
+        """
+        Execute the drop operation for partitions older than the cutoff partition name.
+
+        :param session: SQLAlchemy session.
+        :param table_name: The name of the table with partitions.
+        :param cutoff_partition_name: The cutoff partition name for dropping old partitions.
+        """
+        # Query partitions older than cutoff_partition_name
+        partitions_to_drop = session.execute(
+            text("""
+            SELECT GROUP_CONCAT(PARTITION_NAME) AS partitions
```

Comment:
Using `GROUP_CONCAT` is problematic:
1. It's a MySQL thing. Postgres doesn't support it (it has other alternatives).
2. It's limited to 1024 chars by default (see [here](https://dev.mysql.com/doc/refman/8.4/en/aggregate-functions.html#function_group-concat). Not a big concern for us since we can fit >100 partitions in 1024 chars, but still.

I suggest we don't use this here, but rather can just query the partition names and aggregate later in the Python code.